### ISSUE TITLE
the nested columns

### ISSUE THREAD

**Issue by decaller** [dhg/Skeleton#141]
I just figured out how the nested columns work in skeleton.
CMIIW, the alpha is for the first nested column, and the omega is for the last column (didn't know omega was the last letter of the greek alphabet).

why sekeleton does not use pseudo-class-selectors instead?
is'nt it easier for user instead inserting alpha and omega class everytime?

**Comment by edgarberlinck** [comment]
Good question.

But, if they did'nt, we'll never learm about alpha and omega like you did.

Em terça-feira, 13 de agosto de 2013, Harridi Ilman Tovid escreveu:

> I just figured out how the nested columns work in skeleton.
> CMIIW, the alpha is for the first nested column, and the omega is for the
> last column (didn't know omega was the last letter of the greek alphabet).
> 
> why sekeleton does not use pseudo-class-selectors instead?
> is'nt it easier for user instead inserting alpha and omega class everytime?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/dhg/Skeleton/issues/141
> .

## 

Edgar Muniz Berlinck
http://about.me/edgarberlinck

**Comment by ckjoris** [comment]
this breaks IE compatibility. First-child is sort of supported on ie7, but last-child - only from ie9.

**Comment by decaller** [comment]
Thanks a lot for implementing it. I tried to change it myself but it just didn't work.

**Comment by justindantzer** [comment]
Correct me if I'm wrong, and I might be missing the exact intent, but won't this remove the left margin from all first columns and the right margin from all last columns, whether they're nested inside other columns or not?

For the 960 grid, _.sixteen.columns_ would normally retain `margin-left:10px;` and `margin-right:10px;`, when not also _.alpha_ or _.omega_. With `:first-child` and `:last-child`, the 10px gutter is removed, making the row only 940px wide.

``` html
<div class="container">
    <div class="sixteen columns" />
</div>
```

or

``` html
<div class="container">
    <div class="eight columns" />
    <div class="eight columns" />
</div>
```

versus only removing the gutter margin here, where _.alpha_ and _.omega_ are used

``` html
<div class="container">
    <div class="sixteen columns">
        <div class="eight columns alpha" />
        <div class="eight columns omega" />
    </div>
</div>
```

instead, should the selector be something along the lines of

``` css
.columns .column:first-child, 
.columns .columns:first-child { margin-left: 0; }
.columns .column:last-child, 
.columns .columns:last-child { margin-right: 0; }
```

Just in case you think you'd nest one column inside of one column, you could add

``` css
.column .column:first-child, .column .columns:first-child
```

and

``` css
.column .column:last-child, .column .columns:last-child
```

Also, not specifying the opposite side margin would alleviate the need for

``` css
.column:first-child:last-child, 
.columns:first-child:last-child { margin-left: 0; margin-right: 0; }
```

Another thing to note, when nested columns aren't the only children of the parent, `:first-child` and `:last-child` are no longer helpful, which is unfortunate because I'd prefer this method.

In this example, _.alpha_ and _.omega_ would be necessary

``` html
<div class="container">
    <div class="sixteen columns">
        <div class="four columns alpha">
            <h3>Some heading</h3>
        </div>
        <div class="twelve columns omega">
            <h4>A smaller title</h4>
            <p>Some content</p>
            <div class="five columns alpha">
                <a href="#">Button One</a>
            </div>
            <div class="five columns offset-by-two omega">
                <a href="#">Button Two</a>
            </div>
            <p>Sub-content</p>
        </div>
    </div>
</div>
```

**Comment by rakelley** [comment]
It's worth considering that nesting columns should be strongly discouraged as both unnecessary\* and very problematic if working with grid sizes using percentage instead of pixel widths (which the current version of skeleton does not use, but plenty of off-shoots with extended wide media queries do).

e.g. nesting two eight columns inside a sixteen column gains you nothing over using a row instead, and nesting a 25% column inside a 50% column is not the same as nesting a 25% column inside a 100% container/row.

For my personal variant of Skeleton I use in my current projects, among other changes I've removed the alpha/omega classes and strictly do not nest columns.

*If someone can offer a compelling use case I'm all ears, but refactoring to use rows is the correct choice from both a semantic and organizational standpoint in every situation I've run into.

**Comment by justindantzer** [comment]
Update: _Previous response removed._
I removed my response because I didn't feel the necessity, or lack there of, to nest columns is not directly related to this issue. I'm only proposing a fix for the suggested alteration.

While I generally agree in avoiding nested columns, it is supported in Skeleton. There are weaknesses in both pixel and percentage based frameworks - e.g. fluid-based lack ease of nesting and pixel-based lack fluidity - but they both have their appropriate use cases.

**Comment by dhg** [comment]
Hey all - I'm really sorry I never got to this in a timely fashion originally. I left Skeleton in terrible disrepair for a couple years and without an active developer. I'm back at it though and committed to continuing development. I released v2.0 a week or so ago and am making my way through all the old issues and PRs.

So w/ 2.0, nesting in the grid is not recommended. Since it's a %-based gutter system, that means nested columns will have significantly smaller gutters (can get real tight and weird for nested grids especially at smaller viewport sizes). 

There is a solution to making nested grids work really well. It's the way Bootstrap does it. Essentially it just requires the use of `padding` and `box-sizing` to allow for nestable grid items that maintain perfect spacing and gutters. I experimented with the idea in Skeleton here: https://github.com/dhg/Skeleton/tree/update-grid-test

Advantages of the Bootstrap-esque grid: 
- Nesting with fixed gutters possible

Disadvantages: 
- Creating bordered `.row`s or `.column(s)` on top or bottom requires another nested element to make it the appropriate width
- Content can't live directly in a `.row`, it all must live in `.column(s)` within a `.row` which can sometimes force more markup
- I believe it's a slightly more confusing method given the negative margining, meaning the container is actually wider than it's contents at all times (though I do incur this same confusion as a sub-400px viewport size)

For me, nested grids are actually second to fixed gutters as the primary advantage to the BS style grid.  One easily modifiable padding to change all `.column(s)` gutters is rad. It also means gutter proportions don't scale with browser, but rather are a fixed distance. This is totally subjective to preference and project, but it is a nice option. For now, I'm going to close this issue since I'm sticking with the existing grid for now, but I'm going to mull over the advantages of BS-style grid and will come back to it in the next release. 

Thanks again for opening this and apologies on not getting to it at a sooner date.
